File: examples/CSharpProd/XUnit/XUnitTest.cs
using System;
using System.Threading.Tasks;
using FluentAssertions;
using NBomber.Contracts;
using NBomber.CSharp;
using Xunit;

namespace CSharpProd.XUnit
{
    // in this example we use:
    // - XUnit (https://xunit.net/)
    // - Fluent Assertions (https://fluentassertions.com/)
    // to get more info about test automation, please visit: (https://nbomber.com/docs/test-automation)

    public class XUnitTest
    {
        Scenario BuildScenario()
        {
            var step = Step.Create("simple step", async context =>
            {
                await Task.Delay(TimeSpan.FromSeconds(0.1));
                return Response.Ok(sizeBytes: 1024);
            });

            return ScenarioBuilder.CreateScenario("xunit_hello_world", step);
        }

        [Fact]
        public void Test()
        {
            var scenario = BuildScenario()
                .WithoutWarmUp()
                .WithLoadSimulations(new[]
                {
                    Simulation.KeepConstant(copies: 1, during: TimeSpan.FromSeconds(2))
                });

            var nodeStats = NBomberRunner.RegisterScenarios(scenario).Run();
            var stepStats = nodeStats.ScenarioStats[0].StepStats[0];

            // todo stepStats.OkCount.Should().BeGreaterThan(2);
            stepStats.Ok.Request.RPS.Should().BeGreaterThan(8);
            stepStats.Ok.Latency.Percent75.Should().BeGreaterOrEqualTo(100);
            stepStats.Ok.DataTransfer.MinBytes.Should().Be(1024);
            stepStats.Ok.DataTransfer.AllBytes.Should().BeGreaterOrEqualTo(17408L);
        }
    }
}

File: examples/FSharpProd/XUnit/XUnitTest.fs
module FSharpProd.DataFeed.XUnitTest

open System.Threading.Tasks

open Swensen.Unquote
open Xunit

open NBomber
open NBomber.Contracts
open NBomber.FSharp

// in this example we use:
// - XUnit (https://xunit.net/)
// - Unquote (https://github.com/SwensenSoftware/unquote)
// to get more info about test automation, please visit: (https://nbomber.com/docs/test-automation)

[<Fact>]
let ``XUnit test`` () =

    let step = Step.create("simple step", fun _ -> task {
        do! Task.Delay(milliseconds 100)
        return Response.ok(sizeBytes = 1024)
    })

    let scenario =
        Scenario.create "xunit_hello_world" [step]
        |> Scenario.withoutWarmUp
        |> Scenario.withLoadSimulations [KeepConstant(copies = 1, during = seconds 2)]

    let result = NBomberRunner.registerScenarios [scenario]
                 |> NBomberRunner.run

    match result with
    | Ok nodeStats ->
        let stepStats = nodeStats.ScenarioStats[0].StepStats[0]
        //test <@ stepStats.OkCount > 2 @>
        // todo are all of these ok?
        // TODO - how do you get this?
        //test <@ stepStats.Ok.StatusCodes.Length > 2 @>
        test <@ stepStats.Ok.Request.RPS > 8. @>
        test <@ stepStats.Ok.DataTransfer.Percent75 >= 100 @>
        test <@ stepStats.Ok.DataTransfer.MinBytes = 1024 @>
        test <@ stepStats.Ok.DataTransfer.AllBytes >= 17408L @>

    | Error msg -> failwith msg
